### ISSUE TITLE
Weird bug when expecting Shape2d.Rectangle from PointArrayList

### ISSUE THREAD

**Issue by hooliooo** [korlibs/korma#15]
Hello,

I came across a weird bug when making a Shape2d.Rectangle

```
val points = listOf(Point(x = 1.0, y = 1.0), Point(x = 1.0, y = 2.0), Point(x = 2.0, y = 2.0), Point(x =2.0, y = 1.0))
val t = PointArrayList(points).toShape2d(true)
// t is Shape2d.Polygon instead of Shape2d.Rectangle
```

```
val points2 = listOf(Point(x = 2.0, y = 2.0), Point(x = 2.0, y = 1.0), Point(x = 1.0, y = 1.0), Point(x = 1.0, y = 2.0))
val u = PointArrayList(points2).toShape2d(true)
// u is Shape2d.Polygon instead of Shape2d.Rectangle

val points3 = listOf(Point(x = 2.0, y = 1.0), Point(x = 1.0, y = 1.0), Point(x = 1.0, y = 2.0), Point(x = 2.0, y = 2.0))
val w = PointArrayList(points3).toShape2d(true)
// w is Shape2d.Rectange => correct

val points4 = listOf(Point(x = 1.0, y = 1.0), Point(x = 2.0, y = 1.0), Point(x = 2.0, y = 2.0), Point(x = 1.0, y = 2.0))
val x = PointArrayList(points4).toShape2d(true)
// x is Shape2d.Rectange => correct

```

Shoudn't these all be Shape2d.Rectangle?

**Comment by soywiz** [comment]
I guess that the problem is the direction of the points. Clock-wise points form a rectangle, but counter-clock wise could cause problems, since they might be interpreted as hole polygons. That's why I'm not optimizing those.

Which is your use case for need that to be converted into a rectangle?

**Comment by hooliooo** [comment]
In my use case I am receiving Points that definitely define a Rectangle. Is there a way to reorder the points to go clock-wise to ensure a Shape2d.Rectangle is received?

**Comment by soywiz** [comment]
If  you have a list of points, you can reverse them `listOf(...).reversed()` you can check if any of the two options yields a rectangle and use that one

**Comment by hooliooo** [comment]
Reversing the list did not work. I sorted them from closest to farther to (0,0) then took the closest and farthest points to build the Rectangle. Reversing should have worked though

**Comment by RezMike** [comment]
@soywiz, maybe we should provide a special function `toRectangle` (or `toShape2dRectangle`) that builds a correct Rectangle using provided four points or throws an exception if something is incorrect?

**Comment by soywiz** [comment]
A `List<Point>.tryToRectangle(): Rectangle?` or `List<Point>.toRectangleOrNull(): Rectangle?` could work :+1:

**Comment by RezMike** [comment]
@hooliooo, I've implemented your request here #16,

**Comment by hooliooo** [comment]
Thanks :)
